python/ml4ir/base/features/preprocessing.py

```diff
@@ -53,6 +58,71 @@ def preprocess_text(feature_tensor, remove_punctuation=False, to_lower=False):
     return feature_tensor
 
 
+def get_one_hot_label_vectorizer(feature_info, file_io: FileIO):
+    """
+    Returns a tf function to convert categorical string labels to a one hot encoding.
+
+    Args:
+        feature_info: Dictionary representing the configuration parameters for the specific feature from the FeatureConfig.
+                      See categorical_indicator_with_vocabulary_file, here it is used to read a vocabulary file to
+                      create the one hot encoding.
+        file_io: FileIO required to load the vocabulary file.
+
+    Returns:
+        processed float tensor
+    """
+    # Avoid initialization if there is no vocabulary file stated in feature config wrt to label definition.
+    if not ("args" in feature_info.get("feature_layer_info")) or not (
+            "vocabulary_file" in feature_info.get("feature_layer_info")["args"]):
+        return
+
+    label_str = tf.keras.Input(shape=(1,), dtype=tf.string)
+    label_one_hot = categorical_indicator_with_vocabulary_file(label_str, feature_info, file_io)
+    one_hot_vectorizer = tf.keras.Model(inputs=label_str, outputs=label_one_hot)
```

Comment:
@Ullimague Is it necessary to wrap it as a Keras model? I had to do this in the notebook as a quick workaround. Did you try directly calling the function itself? There might be issues with saving the model(I think). But try it out.